geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/api/QueryPlan.scala

```diff
@@ -285,8 +286,15 @@ object QueryPlan {
     override def schema: SimpleFeatureType = sft
 
     protected def createSerializer: KryoFeatureSerializer = {
+      val hasAttributeLevelVis = sft.getVisibilityLevel == VisibilityLevel.Attribute
       val builder = KryoFeatureSerializer.builder(sft)
```

Comment:
```suggestion
      val builder = KryoFeatureSerializer.builder(sft)
      // note: attribute level vis are serialized into the user data since they can't be held by a single key
```